docs/user-guides/class_notebooks_best_practices.md

```diff
@@ -1,97 +1,134 @@
+- If there are assignment sections requiring students to write or refactor code, test the notebook with the correct solutions. This will alert you to potential issues that you may miss otherwise.
+
+:::{tip} Example
+
+A notebook successfully runs the instructor-provided code cells but crashes the kernel due to insufficient memory when students add new code required to complete the assignment. 
+:::
 
 ---
 
 ## Plan for Students with Poor Internet Access
 
 - Saving a notebook without clearing its output first will increase the file size substantially. 
 - If notebooks are too big and students have poor internet connections, the notebook autosave functionality may fail.
-- Due to the above reasons, it is risky for students to submit their assignments by running notebooks, saving their results, and submitting them afterwards. Students without a strong internet connection may not be able to save and turn in their work in this manner.
-
-#### To avoid issues related to poor internet access, consider following options:
-
-- Allow assignments to be turned in as screenshots pasted into a word processor and converted into pdf.
+- Students without a strong internet connection may not be able to save a completed notebook and turn it in with the output in place.
 
-- Split assignments into 2 notebooks: one for content/examples and another one for assignment. Pass required data structures from the content notebook to the assignment notebook using a [Python pickle](https://docs.python.org/3/library/pickle.html).
+### To avoid these issues, consider allowing students to turn in work with one of the following options:
+1. Print the notebook as a PDF by selecting `Print...` from the `File` menu.
+2. Screenshot relevant portions of the notebook and paste them into a word processor.
     
 ---
     
-## Avoid Changing Directories in Your Code
+## Avoid Changing Directories in a Notebook
 
-#### Why?
-- Users can run Jupyter Notebook code cells in any order. Users can skip over cells and/or re-run previous cells, which can cause unexpected problems. 
-    - *Example:* Consider a Python list that contain a data specific to each day (e.g. temperature, stock price, etc.). To store today's data, you can use `lst.append(todays_data)`. However, you may end with duplicate data in your list if you run this code multiple times since previous output is preserved and thus you are appending `todays_data` multiple times. This may result in breaking code and/or a confusion for students.
-    
-<!-- screenshot here if necessary -->
+### Why?
+- Users can run Jupyter Notebook code cells any number of times and in any order. If a code cell changes directories, there is no way to guarantee that the user will run it in the expected order, which can lead to unexpected behavior.
+
+:::{tip} Example (Don't do this)
```

Comment:
Maybe consider using `{warning}` instead of `{tip}` to make it more obvious its a bad idea.